DOC-2085 fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1)

wuqingjuntg opened the pull request:
Updating documentation for [Load from External Kafka](https://docs.tigergraph.com/tigergraph-server/current/data-loading/load-from-kafka)

[DOC-2085](https://graphsql.atlassian.net/browse/DOC-2085)

[DOC-2085]: https://graphsql.atlassian.net/browse/DOC-2085?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

wuqingjuntg committed:
Revising documentation about data-loading.

pingxieTG commented on modules/data-loading/partials/kafka/kafka-example-loading-job.adoc:
```diff
@@ -2,7 +2,7 @@
 The following is an example loading job from and external Kafka cluster.
 
-[source,php,linenums]
+[source,sql,linenums]
```
Let's use `gsql`

pingxieTG replied:
Did you forget this one?

wuqingjuntg replied:
Interesting, you didn't see `gsql`? From my local I am seeing `gsql` instead of `sql`.

pingxieTG commented on modules/data-loading/partials/kafka/kafka-example-loading-job.adoc:
```diff
@@ -2,7 +2,7 @@
 
-[source,php,linenums]
+[source,sql,linenums]
 .Example loading job for BigQuery
```
Why BigQuery

pingxieTG commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -25,8 +25,9 @@ If the source cluster is configured for SSL or SASL protocols, you need to provi
 Each file must be at the same absolute path on all nodes.
 
-The following configurations are required for admin, producer and consumer. To supply the configuration for the corresponding component, replace `<prefix>` with `source.admin`, `producer`, or `consumer`.
+The following are generic configurations required for admin, producer and consumer. To supply the configuration for the corresponding component, replace `<prefix>` with `source.admin`, `producer`, or `consumer`.
```
Did you tried it? 
* Are `source.admin`, `producer`, or `consumer` all required? 
* Why `admin` requires `source` prefix but others are not?
* How about using `source.cluster`?

wuqingjuntg replied:
I should have mentioned this earlier: 
* We already have a separate instructions on how to set up security (SSL), so user can just follow that instruction to set up SSL, by gadmin enabling a couple of flags, setting certificates etc.;
* This example is simply about avro/mm2, so let's try to make this example as simple as possible;
* This document was provided when SSL is not supported; now given SSL is already supported, we can just recommend user to use SSL if they are concerned about security; 
* All these settings are covered by our current regress test, e.g., cqrs_cluster 101. 
* I would recommend to remove all the wording including SASL from the document. We have a plan to support SASL, but we haven't implemented/supported it yet.

Let me know your thoughts. 
Thanks!

pingxieTG replied:
> All these settings are covered by our current regress test, e.g., cqrs_cluster 101.

* cqrs_cluster 101 is for CRR with SSL, not loading from external Kafka. The SSL setting should be different because the CRR SSL setting is auto-generated.
* In the auto-generated CRR config you wrote, it is using `source.cluster.ssl` instead of `source.admin`, `producer` or `consumer`

wuqingjuntg replied:
- No. cqrs_cluster 101 includes two test cases: test1.sh and test2.sh. test1.sh is for generic mm2; test2.sh is for crr. 
- When I designed SSL for MM2/CRR, connector_manager was used to set up ssl configuration settings to improve the usability. 
- You seemed to ask similar question when you reviewed the SSL PR, here is one of the comments:  https://github.com/tigergraph/cqrs/pull/3001/files#r1432225187 . Basically, Kafka(KafkaConn) maintains precedence of SSL settings: (source or target).cluster.ssl > admin/producer/consumer.ssl etc.. In the design, I just made it simple: all admin/producer/consumer share the same SSL settings to make things simple.

pingxieTG replied:
> Basically, Kafka(KafkaConn) maintains precedence of SSL settings: (source or target).cluster.ssl > admin/producer/consumer.ssl etc.

* Have we explained this to the user? That's why I'm asking to improve the doc.
* Even if we use `admin`, `consumer` and `producer` separately, there's still some mistakes in the doc: why `source.admin` but not no `source.` for `consumer` and `producer`?

pingxieTG commented on modules/data-loading/partials/kafka/kafka-example-loading-job.adoc:
```diff
@@ -2,7 +2,7 @@
 The following is an example loading job from and external Kafka cluster.
 
-[source,php,linenums]
+[source,sql,linenums]
```
I believe this is not the only place using `php`, could you check other code blocks?

pingxieTG replied:
Did you forget this one?

wuqingjuntg committed:
To make it simple, let's support SSL only, and not SASL(including SASL_SSL) in the doc.

wuqingjuntg committed:
sql -> gsql

wuqingjuntg committed:
Addressing comments.

pingxieTG commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -13,61 +13,12 @@ To configure the data source object, the minimum requirement is the address of t
-
-* If the source cluster uses SASL, you need to upload the keytab of each Kerberos principal to every node of your TigerGraph cluster at the same absolute path.
 * If the source cluster uses SSL, see our documentation xref:tigergraph-server:data-loading:kafka-ssl-security-guide.adoc[]
-* If the source cluster uses SASL *and* SSL, you need to upload the keytab of each Kerberos principal, as well as the key store and truststore to every node of your TigerGraph cluster.
-Each file must be at the same absolute path on all nodes.
-
-The following configurations are required for admin, producer and consumer. To supply the configuration for the corresponding component, replace `<prefix>` with `source.admin`, `producer`, or `consumer`.
```
Why remove them?

pingxieTG commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -13,61 +13,12 @@ To configure the data source object, the minimum requirement is the address of t
-If the source cluster is configured for SSL or SASL protocols, you need to provide the following SSL/SASL credentials in order to communicate with the source cluster.
-
-* If the source cluster uses SASL, you need to upload the keytab of each Kerberos principal to every node of your TigerGraph cluster at the same absolute path.
 * If the source cluster uses SSL, see our documentation xref:tigergraph-server:data-loading:kafka-ssl-security-guide.adoc[]
```
https://graphsql.atlassian.net/browse/DOC-1972?focusedCommentId=152682 see comment for `kafka-ssl-security-guide.adoc`.
* SSL for internal Kafka
* SSL for CRR
* SSL for loading from external Kafka

They should belong to 3 different categories(under 3 different pages) instead of putting them together.

wuqingjuntg replied:
- Let's clarify here. We're not going to cover SSL for CRR here right?
- For internal and external Kafka, I don't see major difference. The "internal" Kafka is set up for testing loading from regular Kafka cluster including external Kafka. Let me know if there's any understanding gap. We can have a short discussion in Tuesday's meeting and get consensus about this with the team.  ( At the moment I still have a couple of JIRA tickets to handle, which also have high priorities).
- I am removing the SASL and the SASL/SSL parameters, because:  without SSL, SASL is not secured. SASL is authentication method, but without encryption, the connection is not secured. So we should simply recommend the users to enable regular SSL to secure the connection between Kafka clusters. User doesn't need to use both SASL + SSL. SSL is enough. And we didn't officially support SASL yet.
- My suggestion is that, for all the configuration/settings in the DOC, we should have already supported the feature and also have test cases covering those configs.

cc @arunramasami , FYI. Any comments regarding this DOC change I am making?

wuqingjuntg committed:
Merge branch '4.1.o' into DOC-2085

wuqingjuntg committed:
Resolving code review comments.

romansod commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -25,9 +25,17 @@ If the source cluster is configured for SSL or SASL protocols, you need to provi
+If source and target clusters have different SSL settings, to make things simple, users can simply set cluster wide SSL configs, e.g., "target.cluster.ssl.truststore.password=/password/for/trust/store", instead of "target.cluster.producer.ssl.trust.password=/password/for/trust/store".
+
+To supply the configuration for the corresponding component, replace `<prefix>` with `source(/or target).cluster`, `source(or target).cluster.admin(or producer, consumer)`, `admin`, `producer`, or `consumer`.
 For example, to specify `GSSAPI` as the SASL mechanism for consumer, include `"consumer.sasl.mecahnism": "GSSAPI"` in the data source configuration.
```
typo -> mechanism

romansod commented on modules/data-loading/partials/load-part5-monitor-and-manage.adoc:
```diff
@@ -3,7 +3,7 @@
 When a loading job starts, the GSQL server assigns it a job ID and displays it for the user to see.
```
This entire section can be replaced by 

https://docs.tigergraph.com/gsql-ref/4.1/ddl-and-loading/managing-loading-job

romansod commented on modules/data-loading/partials/load-part1-intro-and-schema.adoc:
```diff
@@ -6,7 +6,7 @@ We will call out whether a particular step is common for all loading or specific
 == Example Schema
```
line 1

remove comma after 'defined a graph schema'

romansod replied:
line 3 replace with:

The steps for loading from local files, cloud storage, or any other supported sources are similar. We will distinguish where steps are common for all loading jobs or are specific to a particular data source or loading mode.

romansod commented on modules/data-loading/partials/load-part1-intro-and-schema.adoc:
```diff
@@ -6,7 +6,7 @@ We will call out whether a particular step is common for all loading or specific
 == Example Schema
 This example uses part of the LDBC_SNB schema:
```
Remove line 7, we have restated multiple times that this is an example schema, and twice that it is from LDBC_SNB

romansod commented on modules/data-loading/partials/load-part2-create-data-source.adoc:
```diff
@@ -8,13 +8,13 @@ Inline mode is required when creating data sources for TigerGraph Cloud instance
 
```
Line 3

Is 'data source code' the intended identifier? Or should this have been 'data source object' ?

romansod replied:
Line 7

Wrap this as a note

```
[NOTE]
====
Inline mode is required when creating data sources for TigerGraph Cloud instances.
====
```

romansod commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -13,8 +13,8 @@ To configure the data source object, the minimum requirement is the address of t
 .Data source configuration for external Kafka
```
Line 7

In addition, users can utilize external sources to provide configurations for Kafka connectors. This includes files, Vault, and environment variables. See xref:tigergraph-server:data-loading:externalizing-kafka-configs.adoc[].

romansod commented on modules/data-loading/partials/load-part2-create-data-source.adoc:
```diff
@@ -8,13 +8,13 @@ Inline mode is required when creating data sources for TigerGraph Cloud instance
 USE GRAPH ldbc_snb
 CREATE DATA_SOURCE s1 = "ds_config.json" FOR GRAPH ldbc_snb
 
 Older versions of TigerGraph required a keyword after `DATA_SOURCE` such as `STREAM` or `KAFKA`.
```
line 15

wrap this as a note
```
[NOTE]
====
Older versions of TigerGraph required a keyword after `DATA_SOURCE` such as `STREAM` or `KAFKA`.
====
```

romansod commented on modules/data-loading/partials/load-part2-create-data-source.adoc:
```diff
@@ -24,7 +24,7 @@ key: <value>
 String literals can be enclosed with a double quote `"`, triple double quotes `"""`, or triple single quotes `'''`.
 Double quotes `"` in the JSON can be omitted if the key name does not contain a colon `:` or comma `,`.
```
Line 24, 25

This should be reformatted, perhaps into bullet points with a sub header

### String Literals
String literals can be represented according to the following options
- Enclosed with double quote `"`.
- Enclosed with triple double quotes `"""`.
- Enclosed with triple single quotes `'''`.
In the case of JSON that does not contain a colon `:` or a comma `,` the double quotes `"` can be omitted

romansod commented on modules/data-loading/partials/load-part2-create-data-source.adoc:
```diff
@@ -24,7 +24,7 @@ key: <value>
+[source,gsql]
 .Alternate quote syntax for inline JSON data
 CREATE DATA_SOURCE s1 = """{
 "type": "<type>",
```
line 34

Both a period `.` and `_` can be used for separation in the specified key name. Example: `first.second` or `first_second`.

romansod reviewed (commented):
Leaving some inline notes to implement later

pingxieTG commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -25,9 +25,17 @@ If the source cluster is configured for SSL or SASL protocols, you need to provi
 Each file must be at the same absolute path on all nodes.
 
-The following configurations are required for admin, producer and consumer. To supply the configuration for the corresponding component, replace `<prefix>` with `source.admin`, `producer`, or `consumer`.
+The following configurations are required for admin, producer and consumer. Basically Kafka allows SSL settings overriding, it respects security settings in precedence order: generic.ssl.setting < source/target.cluster.ssl.setting < admin/producer/consumer.ssl.setting.
+
+If both source and target clusters are sharing the same SSL settings, user can set generic settings for both source/target clusters and all the rols(admin/producer/consumer). For example, user can set "ssl.keystore.location=/path/to/key/store" instead of "source.cluster.ssl.keystore.location=/path/to/key/store", or "admin.ssl.keystore.location=/path/to/key/store", or even "source.cluster.admin.ss ...
+
+If source and target clusters have different SSL settings, to make things simple, users can simply set cluster wide SSL configs, e.g., "target.cluster.ssl.truststore.password=/password/for/trust/store", instead of "target.cluster.producer.ssl.trust.password=/password/for/trust/store".
+
+To supply the configuration for the corresponding component, replace `<prefix>` with `source(/or target).cluster`, `source(or target).cluster.admin(or producer, consumer)`, `admin`, `producer`, or `consumer`.
```
Actually users only need to configure for `consumer.xxx`(consume from source kafka cluster), `source.admin.xxx`(list topics of the source kafka cluster), `producer.xxx`(produce the read offset to source kafka cluster).

We'd better give at least 4 examples for PLAINTEXT, SSL, SASL_PLAINTEXT, SASL_SSL, which is given by `<prefix>.security.protocol`:
* Plaintext(no ssl and sasl), in this case, no need to configure consumer, source.admin and producer:
```
{
    "type": "mirrormaker",
    "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>"
}
```
* SSL only(configure `<prefix>.security.protocol` and `<prefix>.ssl.xxx`)
```
{
    "type": "mirrormaker",
    "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>",

    "consumer.security.protocol": "SSL",
    "consumer.ssl.endpoint.identification.algorithm": "none",
    "consumer.ssl.keystore.location": "/path/to/client.keystore.jks",
    "consumer.ssl.keystore.password": "******",
    "consumer.ssl.key.password": "******",
    "consumer.ssl.truststore.location": "/path/to/client.truststore.jks",
    "consumer.ssl.truststore.password": "******",

    "source.admin.security.protocol": "SSL",
    "source.admin.ssl.endpoint.identification.algorithm": "none",
    "source.admin.ssl.keystore.location": "/path/to/client.keystore.jks",
    "source.admin.ssl.keystore.password": "******",
    "source.admin.ssl.key.password": "******",
    "source.admin.ssl.truststore.location": "/path/to/client.truststore.jks",
    "source.admin.ssl.truststore.password": "******",

    "producer.security.protocol": "SSL",
    "producer.ssl.endpoint.identification.algorithm": "none",
    "producer.ssl.keystore.location": "/path/to/client.keystore.jks",
    "producer.ssl.keystore.password": "******",
    "producer.ssl.key.password": "******",
    "producer.ssl.truststore.location": "/path/to/client.truststore.jks",
    "producer.ssl.truststore.password": "******"
}
```
* SASL_PLAINTEXT(configure `<prefix>.security.protocol` and `<prefix>.sasl.xxx`)
```
{
    "type": "mirrormaker",
    "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>",

    "consumer.security.protocol": "SASL_PLAINTEXT",
    "consumer.sasl.mechanism": "GSSAPI/PLAIN/etc.",
    "consumer.sasl.jaas.config": "<jaas_conf_content>",
    "consumer.sasl.client.callback.handler.class": "optional, need to put third party jar under `$(gadmin config get System.Approot)/kafka/libs/`",

    "source.admin.security.protocol": "SASL_PLAINTEXT",
    "source.admin.sasl.mechanism": "GSSAPI/PLAIN/etc.",
    "source.admin.sasl.jaas.config": "<jaas_conf_content>",
    "source.admin.sasl.client.callback.handler.class": "optional, need to put third party jar under `$(gadmin config get System.Approot)/kafka/libs/`",

    "producer.security.protocol": "SASL_PLAINTEXT",
    "producer.sasl.mechanism": "GSSAPI/PLAIN/etc.",
    "producer.sasl.jaas.config": "<jaas_conf_content>",
    "producer.sasl.client.callback.handler.class": "optional, need to put third party jar under `$(gadmin config get System.Approot)/kafka/libs/`"
}
```
* SASL_SSL(configure `<prefix>.security.protocol`, `<prefix>.sasl.xxx` and `<prefix>.ssl.xxx`)
```
{
    "type": "mirrormaker",
    "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>",

    "consumer.security.protocol": "SASL_SSL",
    "consumer.sasl.mechanism": "GSSAPI/PLAIN/etc.",
    "consumer.sasl.jaas.config": "<jaas_conf_content>",
    "consumer.sasl.client.callback.handler.class": "optional, need to put third party jar under `$(gadmin config get System.Approot)/kafka/libs/`",
    "consumer.ssl.endpoint.identification.algorithm": "none",
    "consumer.ssl.keystore.location": "/path/to/client.keystore.jks",
    "consumer.ssl.keystore.password": "******",
    "consumer.ssl.key.password": "******",
    "consumer.ssl.truststore.location": "/path/to/client.truststore.jks",
    "consumer.ssl.truststore.password": "******",

    "source.admin.security.protocol": "SASL_PLAINTEXT",
    "source.admin.sasl.mechanism": "GSSAPI/PLAIN/etc.",
    "source.admin.sasl.jaas.config": "<jaas_conf_content>",
    "source.admin.sasl.client.callback.handler.class": "optional, need to put third party jar under `$(gadmin config get System.Approot)/kafka/libs/`",
    "source.admin.ssl.endpoint.identification.algorithm": "none",
    "source.admin.ssl.keystore.location": "/path/to/client.keystore.jks",
    "source.admin.ssl.keystore.password": "******",
    "source.admin.ssl.key.password": "******",
    "source.admin.ssl.truststore.location": "/path/to/client.truststore.jks",
    "source.admin.ssl.truststore.password": "******",

    "producer.security.protocol": "SASL_PLAINTEXT",
    "producer.sasl.mechanism": "GSSAPI/PLAIN/etc.",
    "producer.sasl.jaas.config": "<jaas_conf_content>",
    "producer.sasl.client.callback.handler.class": "optional, need to put third party jar under `$(gadmin config get System.Approot)/kafka/libs/`",
    "producer.ssl.endpoint.identification.algorithm": "none",
    "producer.ssl.keystore.location": "/path/to/client.keystore.jks",
    "producer.ssl.keystore.password": "******",
    "producer.ssl.key.password": "******",
    "producer.ssl.truststore.location": "/path/to/client.truststore.jks",
    "producer.ssl.truststore.password": "******"
}
```

romansod committed:
Merge branch '4.1' into DOC-2085

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Taking over ticket, adding formatting, spelling, and content fixes

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Taking over ticket, adding formatting, spelling, and content fixes

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Additional formatting and better representation of security protocols, as well as examples

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Additional formatting

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Additional formatting

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Additional formatting

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Additional formatting

romansod commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -13,62 +13,239 @@ To configure the data source object, the minimum requirement is the address of t
+  "consumer.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+
+  "source.admin.security.protocol": "SASL_PLAINTEXT",
+  "source.admin.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
```
@pingxieTG  is the `<prefix>.sasl.jaas.config` only needed for third party  security protocol? Or would it also be necessary for the default?

pingxieTG replied:
`sasl.jaas.config` is required for all mechanisms including the built-ins.
For example: https://docs.confluent.io/platform/current/security/authentication/sasl/gssapi/overview.html#clients
Here the GSSAPI has the following config:
```
sasl.jaas.config=com.sun.security.auth.module.Krb5LoginModule required \
    useKeyTab=true \
    storeKey=true \
    keyTab="/etc/security/keytabs/kafka_client.keytab" \
    principal="kafkaclient1@EXAMPLE.COM";
```

romansod commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -13,62 +13,239 @@ To configure the data source object, the minimum requirement is the address of t
+[source,json,linenum]
+.Example SASL Configuration with third party mechanism
+----
+{
```
@pingxieTG My intention here is to give a proper example of a third party security mechanism for SASL. What optional arguments do we want to show here instead of <GSSAPI/PLAIN/etc> ?

I think the other fields are okay as generics as they are explained in detail in the table of parameters

pingxieTG replied:
Pls check this example: https://github.com/aws/aws-msk-iam-auth?tab=readme-ov-file#configuring-a-kafka-client-to-use-aws-iam-with-aws_msk_iam-mechanism

romansod committed:
[DOC-2085] fix(dataloading): improve doc for Load from External Kafka (3.9?-4.1);
Re-added jaas config

romansod committed:
Merge branch '4.1' into DOC-2085

victorleeTG commented on modules/data-loading/pages/data-loading-overview.adoc:
```diff
@@ -46,7 +46,7 @@ TigerGraph uses the same workflow for both local file and Kafka Connect loading:
 Data is always loading to exactly one graph (though that graph could have global vertices and edges which are shared with other graphs). For example:
 +
-[source,php]
+[source,gsql]
```
The doc editor doesn't have a style guide for gsql, so this will have no effect.  I support it is useful for forward compatibility. We might be able to introduce gsql code styling rules in the future.

victorleeTG commented on modules/data-loading/partials/kafka/kafka-data-source-details.adoc:
```diff
@@ -4,7 +4,7 @@ The TigerGraph connector to external Kafka sources makes use of https://cwiki.ap
 [NOTE]
 ====
-Users can additionally utilize external sources, including files, Vault, and environment variables, to provide configurations for Kafka connectors. See xref:tigergraph-server:data-loading:externalizing-kafka-configs.adoc[].
+In addition, users can utilize external sources to provide configurations for Kafka connectors. This includes files, Vault, and environment variables. See xref:tigergraph-server:data-loading:externalizing-kafka-configs.adoc[].
```
```suggestion
In addition, users can utilize external sources to provide configurations for Kafka connectors. These sources include files, Vault, and environment variables. See xref:data-loading:externalizing-kafka-configs.adoc[].
```

victorleeTG commented on modules/data-loading/partials/load-part1-intro-and-schema.adoc:
```diff
@@ -1,10 +1,8 @@
-After you have xref:gsql-ref:ddl-and-loading:defining-a-graph-schema.adoc[defined a graph schema], you can create a loading job, specify your data sources, and run the job to load data.
+After you have xref:gsql-ref:ddl-and-loading:defining-a-graph-schema.adoc[defined a graph schema] you can create a loading job, specify your data sources, and run the job to load data.
```
```suggestion
After you have xref:{page-component-version}@gsql-ref:ddl-and-loading:defining-a-graph-schema.adoc[defined a graph schema], you can create a loading job, specify your data sources, and run the job to load data.
```

victorleeTG commented on modules/data-loading/partials/load-part2-create-data-source.adoc:
```diff
@@ -1,28 +1,36 @@
+* Enclosed with double quote `"`.
+* Enclosed with triple double quotes `"""`.
+* Enclosed with triple single quotes `'''`.
+In the case of JSON that does not contain a colon `:` or a comma `,` the double quotes `"` can be omitted
```
```suggestion
In the case of JSON that does not contain a colon `:` or a comma `,` the double quotes `"` can be omitted.
```

victorleeTG commented on modules/data-loading/partials/load-part2-create-data-source.adoc:
```diff
@@ -31,4 +39,4 @@ CREATE DATA_SOURCE s1 = """{
 
-Key names accept a separator of either a period `.` or underscore `_`, so for example, `key_name` and `key.name` are both valid key names.
\ No newline at end of file
+Both a period `.` and `_` can be used for separation in the specified key name. Example: `first.second` or `first_second`.
```
```suggestion
Either a period `.` or `_` can be used for separation in the specified key name. Example: `first.second` or `first_second`.
```

victorleeTG commented on modules/data-loading/partials/load-part1-intro-and-schema.adoc:
```diff
@@ -1,10 +1,8 @@
 
-The steps are similar whether you are loading from local files, from cloud storage, or any of the other supported sources.
-We will call out whether a particular step is common for all loading or specific to a data source or loading mode.
+The steps for loading from local files, cloud storage, or any other supported sources are similar. We will distinguish where steps are common for all loading jobs, or are specific to a particular data source or loading mode.
```
```suggestion
The steps for loading from local files, cloud storage, or any other supported sources are similar.
We will call out whether a particular step is common for all loading or specific to a data source or loading mode.
```
Rejecting the changes to the second sentence. "Call out" is more semantically appropriate than "distinguish". We are calling attention.  Also, it is better asciidoc style for each sentence to be on a separate line, to simplify revisions.

victorleeTG commented on modules/data-loading/partials/load-part5-monitor-and-manage.adoc:
```diff
@@ -1,32 +1,5 @@
 == Manage and monitor your loading job
 
```
```suggestion
When a loading job starts, the GSQL server assigns it a job ID and displays it for the user to see.
There are four key commands to monitor and manage loading jobs:

[source,gsql]
----
SHOW LOADING STATUS job_id|ALL
ABORT LOADING JOB job_id|ALL
RESUME LOADING JOB job_id
SHOW LOADING ERROR job_id
----

`SHOW LOADING STATUS` shows the current status of either a specified loading job or all current jobs, this command should be within the scope of a graph:

[source,gsql]
GSQL > USE GRAPH graph_name
GSQL > SHOW LOADING STATUS ALL

For each loading job, the above command reports the following information:

* Loading status
* Loaded lines/Loaded objects/Error lines
* Average loading speed
* Size of loaded data
* Duration
When inspecting all current jobs with `SHOW LOADING STATUS ALL`, the jobs in the `FINISHED` state will be omitted as they are considered to have successfully finished. You can use `SHOW LOADING STATUS job_id` to check the historical information of finished jobs. If the report for this job contains error data, you can use `SHOW LOADING ERROR job_id` to see the original data that caused the error.

```
rejecting the mass deletion.

victorleeTG commented on modules/data-loading/partials/load-part5-monitor-and-manage.adoc:
```diff
@@ -1,32 +1,5 @@
-
-When inspecting all current jobs with `SHOW LOADING STATUS ALL`, the jobs in the `FINISHED` state will be omitted as they are considered to have successfully finished. You can use `SHOW LOADING STATUS job_id` to check the historical information of finished jobs. If the report for this job contains error data, you can use `SHOW LOADING ERROR job_id` to see the original data that caused the error.
-
 See xref:gsql-ref:ddl-and-loading:managing-loading-job.adoc[Managing and Inspecting a Loading Job] for more details.
```
```suggestion
See xref:{page-component-version}@gsql-ref:ddl-and-loading:managing-loading-job.adoc[Managing and Inspecting a Loading Job] for more details.
```

victorleeTG committed:
Apply suggestions from code review

Victor's edits to the PR: Rejected the mass deletion of content from partial load-part5-monitor-and-manage.adoc.  Other changes are (1) wording tweaks or (2) the correct way to handle xrefs within a component or to another component.
